src_code/src/controllers/services/Authentication/SignUp.js

```diff
@@ -0,0 +1,163 @@
+    try{ 
+        /** Validate if Sign Up information from request body is valid */
+        const errors = await SignUpValidation(res, req);
+        if (errors) {
+            return errors;
+        } 
+        
+        /** Destructure variables from the request body */
+        const {userRole, firstName, middleName, lastName, emailAddress, accountPassword } = req;
+        
+        /** Handle if emtpy studentId, it should be null instead of empty string */
+        if(req.studentId.length === 0 || req.studentId === "") {
+            req.studentId = null;
+        } else {
+            /** If there is studentId, trim trailing leading spaces */
+            req.studentId = req.studentId.trim();
+        }
+
+        /** Generate salt for encrypting password */
+        const salt = await bcrypt.genSalt(10); 
+
+        /** Encrypt the password */
+        const encryptedPassword = await bcrypt.hash(accountPassword, salt);
+
+        /** Prepare data we insert to the table */
+        const insertData = {
+            USER_ROLE: userRole.trim(),
+            FIRST_NAME: firstName.trim(),
+            MIDDLE_NAME: middleName.trim(),
+            LAST_NAME: lastName.trim(),
+            STUDENT_ID: req.studentId,
+            EMAIL_ADDRESS: emailAddress.trim(),
+            ACCOUNT_PASSWORD: encryptedPassword,
+        }
+
+        /** Inserting 'insertData' into the table */
+        await db("user_info").insert(insertData);
+
+        /** Return 'Sign up successfully' message after insert to table successfully */
+        return responseBuilder.BuildResponse(res, 200, {
+            message: "Sign up successfully."
+        });
+    } catch(error){ 
+        /** If an unexpected server error occurs while signing up after passing validation, return log and error. */
+        console.log("ERROR: An error occurred while signing up: ", error);
+        return responseBuilder.ServerError(res, "Sorry, there was an error while signing up.");
+    }
+}
+
+/**
+ * Handle sign up validation (TODO: Have not verify if email address is faculty or admin yet).
+ * @param {object} res - the response object for the HTTP request.
+ * @param {object} req - the body of the request containing sign up details.
+ * @returns {object} res - if failed validation, send back to client bad request response, otherwise null.
+ */
+async function SignUpValidation(res, req) {
+    try {
+        /** Declare a email pattern ends with @spu.edu (regex) */
+        const emailPattern = /.+@spu\.edu$/;
+
+        /** Declare a studentID pattern */
+        const studentIDPattern = /^900\d{6}$/;
+
+
+        /** Destructure variables from the request body */
+        const {userRole, firstName, lastName, studentId, emailAddress, accountPassword } = req;
+
+        /** We check for userRole and studentId even though they are optional because we expect to have them in request body */
+        if(!userRole || !firstName || !lastName || !emailAddress || !accountPassword) {
```

Comment:
nit: same as above, I believe you can refactor some of these if statements to be more concise. Here's what Sonarlint's showing me:
<img width="736" alt="complex-3" src="https://github.com/tranleduyan/ETC-Backend/assets/97055517/08e3b447-0680-4653-88b6-cd1b03778182">